ui/StatusQ/src/StatusQ/Components/StatusMessageHeader.qml

```diff
@@ -133,64 +139,93 @@ Item {
+                            return "tiny/message/sent"
+                        case StatusMessage.OutgoingStatus.Sending:
+                            return "tiny/pending"
+                        case StatusMessage.OutgoingStatus.Expired:
+                            return "tiny/tiny-exclamation"
+                        default:
+                            return ""
+                        }
+                    }
+                }
+                Loader {
+                    active: root.showOutgointStatusLabel
+                    asynchronous: true
+                    sourceComponent: StatusBaseText {
+                        Layout.alignment: Qt.AlignVCenter
+                        color: d.outgoingStatusColor
+                        font.pixelSize: Theme.asideTextFontSize
+                        text: {
+                            if (root.resendError != "")
+                                return qsTr("Failed to resend: %1").arg(root.resendError)
+                            switch (root.outgoingStatus) {
+                            case StatusMessage.OutgoingStatus.Delivered:
+                                return qsTr("Delivered")
+                            case StatusMessage.OutgoingStatus.Sent:
+                                return qsTr("Sent")
+                            case StatusMessage.OutgoingStatus.Sending:
+                                return qsTr("Sending")
+                            case StatusMessage.OutgoingStatus.Expired:
+                                return qsTr("Sending failed")
+                            default:
+                                return ""
+                            }
+                        }
+                    }
+                }
             }
         }
 
         Loader {
-            id: sendingInProgressLoader
-            active: showSendingLoader && !!timestampText.text
+            id: resendButtonLoader
+            active: root.showOutgointStatusLabel && d.expired
             asynchronous: true
-            sourceComponent: StatusBaseText {
-                id: sendingInProgress
-                verticalAlignment: Text.AlignVCenter
-                color: Theme.palette.baseColor1
-                font.pixelSize: Theme.tertiaryTextFontSize
-                text: qsTr("Sending...") // TODO replace this with the required design
-            }
-        }
-
-        Component {
-            id: dotComponent
-            StatusBaseText {
-                id: dot
-                verticalAlignment: Text.AlignVCenter
-                font.pixelSize: Theme.asideTextFontSize
-                color: Theme.palette.baseColor1
-                text: "•"
+            sourceComponent: StatusButton {
+                Layout.fillHeight: true
+                verticalPadding: 0
+                horizontalPadding: 5
+                size: StatusBaseButton.Tiny
+                type: StatusBaseButton.Warning
+                font.pixelSize: 11
```

Comment:
I'd even make this smaller, seems too big in the video

Reply:
It was actually smaller, but somehow was lost by me 😄 
Fixed